docs/_examples/vcenter-events.md

```diff
@@ -0,0 +1,223 @@
+            "log.keepOld"="10";
+            "log.rotateSize"="100000"
+            "RemoteDisplay.maxConnections"="2";
+            "RemoteDisplay.vnc.enabled"="false";  
+        
+        }
+        if ($DebugPreference -eq "Inquire") {
+            Write-Output "VM Hardening Options:"
+            $ExtraOptions | Format-Table -AutoSize
+        }
+        
+        $VMConfigSpec = New-Object VMware.Vim.VirtualMachineConfigSpec
+        
+        Foreach ($Option in $ExtraOptions.GetEnumerator()) {
+            $OptionValue = New-Object VMware.Vim.optionvalue
+            $OptionValue.Key = $Option.Key
+            $OptionValue.Value = $Option.Value
+            $VMConfigSpec.extraconfig += $OptionValue
+        }
+        ForEach ($VM in $VMs){
+                $VMv = Get-VM $VM | Get-View
+            $state = $VMv.Summary.Runtime.PowerState
+            Write-Output "...Starting Reconfiguring VM: $VM "
+            $TaskConf = ($VMv).ReconfigVM_Task($VMConfigSpec)
+                if ($state -eq "poweredOn") {
+                    Write-Output "...Migrating VM: $VM "
+                    $TaskMig = $VMv.MigrateVM_Task($null, $_.Runtime.Host, 'highPriority', $null)
+                    }
+            }
+        }
+}
+    
+function handle($context, $payload) {
+    [void](Set-PowerCLIConfiguration -InvalidCertificateAction ignore -Confirm:$false)
+
+    $username = $context.secrets.username
+    $password = $context.secrets.password
+    $hostname = $context.secrets.host
+    $vmName = $payload.metadata.vm_name
+
+    # Connect to vSphere
+    Write-Host "Checking VC Connection is active"
+    if (-not $global:defaultviservers) {
+        Write-Host "Connecting to $hostname"
+        $server = connect-viserver -server $hostname -User $username -Password $password
+    } else {
+        Write-Host "Already connected to $hostname"
+    }
+
+    Write-Host "Get Virtual Machine By name"
+    $vm = Get-VM -Name $vmName
+
+    Write-Host "Security Harden our VM"
+    $vm | Apply-Hardening 
+    
+    return "success"
+}
+```
+
+2. Create Dispatch function using previously created image and secret (assuming the function file is called hardenvm.ps1):
+```
+$ dispatch create function harden-vm hardenvm.ps1 --image=powershell-powercli --secret vsphere
+Created function: harden-vm
+```
+
+If you have a VM in your vSphere already, you can test the function before wiring it to vSphere events by running it manually:
+
+```
+$ dispatch exec harden-vm --wait --input '{"metadata": {"vm_name": "myvm"}}' | jq -r '.output'
+[
+  "...Starting Reconfiguring VM: myvm ",
+  "success"
+]
+``` 
+
+Replace `myvm` with the name of your vm. 
+
+# Create an event driver
+In order to receive vSphere events we need to create an event driver.
+
+1. Create a vcenter event driver type which registers the docker image implementing the driver:
+```
+$ dispatch create event-driver-type vcenter dispatchframework/dispatch-events-vcenter:solo-auth
```

Comment:
We should figure out some tagging pattern that's consistent across the event drivers.